### DISCUSSION TITLE
Do we have a ready make docker image for CORTX?

### DISCUSSION THREAD

**Discussion by kiwionly2** [Seagate/cortx#712] [Q&A] [answered]
Hi, 

I was reading [Release Build Creation](../blob/main/doc/Release_Build_Creation.rst), and seem like need to make docker image our self.

Just wondering do we got any ready make docker image can just load and run immediately ? 

I was plan to run the docker image on Windows 10.

Regards,
kiwi

**Comment by mukul-seagate11** [comment]
Thanks for opening this issue. A contributor should be by to give feedback soon. In the meantime, please check out the [https://github.com/Seagate/cortx/blob/main/CONTRIBUTING.md](https://github.com/Seagate/cortx/blob/main/CONTRIBUTING.md) and explore other https://github.com/Seagate/cortx/

**Comment by mukul-seagate11** [comment]
@kiwionly2, You can incorporate OVA image as updated in [Release Virtual Appliance Releases · Seagate/cortx (github.com)](https://github.com/Seagate/cortx/releases/tag/VA) which is a ready made solution based on CORTX software where you can import the same on VMware Workstation etc. which is designed from test/dev environment perspective.

Reference documentation:

[cortx/CORTX\_on\_Open\_Virtual\_Appliance.rst at main · Seagate/cortx (github.com)](https://github.com/Seagate/cortx/blob/main/doc/CORTX_on_Open_Virtual_Appliance.rst)

[cortx/Preboarding\_and\_Onboarding.rst at main · Seagate/cortx (github.com)](https://github.com/Seagate/cortx/blob/main/doc/Preboarding_and_Onboarding.rst)

**Comment by kiwionly2** [comment]
I mean something like https://hub.docker.com/_/rabbitmq for CORTX

Which I could just run 

```
$ docker run -d --hostname my-rabbit --name some-rabbit rabbitmq:3
```

**Comment by justinzw** [comment]
Hi @kiwionly2 thank you for your interest and question. Unfortunately at this time we do not have the ability to run CORTX within a docker container. 

As @mukul-seagate11 mentioned we do have the ability to run CORTX within a VM 
https://github.com/Seagate/cortx/blob/main/doc/CORTX_on_Open_Virtual_Appliance.rst

Would this work for you? I'm going to close this issue for now as it's something we are not supporting. But will certainly re-open if we have any updates on when we will provide docker support.

Also let me know if I misunderstood your question and you are referring to something else.

**Comment by justinzw** [comment]
@kiwionly2  I've converted this to a discussion as I believe other people in our community might have the same request.

+ @johnbent @gregnsk we can use this discussion thread to provide updates on docker support.

**Reply by sandkumaCode** [reply]
do we have a docker container would be interested in running cortx s3 server as  microservice

**Reply by johnbent** [reply]
Hello @sandkumaCode , have you looked at our minikube and aws and k8s deployment guides?
https://github.com/Seagate/cortx-k8s/blob/integration/README.md#kubernetes-reference-deployments
https://github.com/Seagate/cortx-k8s/search?q=docker

**Reply by sandkumaCode** [reply]
thanks checking docker install part...  thanks for the pointer

**Comment by johnbent** [comment]
@justinzw , looks like maybe we need to update the "Answer" here.

**Comment by johnbent** [comment, marked as answer]
Hello @sandkumaCode , have you looked at our minikube and aws and k8s deployment guides?
https://github.com/Seagate/cortx-k8s/blob/integration/README.md#kubernetes-reference-deployments
https://github.com/Seagate/cortx-k8s/search?q=docker
